src/game/gameModes.ts

```diff
@@ -2,22 +2,57 @@ import { initializePlayers } from "../database/db_sequelize";
 import { balanceTeams } from "./balance";
 import { Player } from "./player";
 
+export enum GameModesStandard {
+	CS = 'cs',
+	DOTA = 'dota',
+	VALORANT = 'valorant',
+	TEST = 'test'
+}
+
+export enum GameModes1v1 {
+	CS1v1 = 'cs1v1',
+	DOTA1v1 = 'dota1v1',
+	TEST1v1 = 'test1v1',
+}
+
+export enum GameModeRatings {
+	TRIVIA = 'trivia',
+	TRIVIA_TEMP = 'trivia_temp',
+}
+
+export type GameModesType = 
+	GameModesStandard.CS |
+	GameModesStandard.DOTA |
+	GameModesStandard.VALORANT |
+	GameModesStandard.TEST |
+	GameModes1v1.CS1v1 | 
+	GameModes1v1.DOTA1v1 |
+	GameModes1v1.TEST1v1 |
+	GameModeRatings.TRIVIA |
+	GameModeRatings.TRIVIA_TEMP
+
+export class GameModes {
+
+	// TODO: Move to static functions in here
+}
+
 // Default choices is the first indexed mode
-export const modesGame = ['cs', 'dota', 'valorant', 'test'];
-export const modes1v1 = ['cs1v1', 'dota1v1', 'test1v1'];
-export const modesRatings = ['trivia', 'trivia_temp']; // Trivia temp is used for ongoing games
+// TODO: Enum to list - Didn't find good way to do it since enum return keys and values
+export const modesGame = [GameModesStandard.CS, GameModesStandard.DOTA, GameModesStandard.VALORANT, GameModesStandard.TEST];
+export const modes1v1 = [GameModes1v1.CS1v1, GameModes1v1.DOTA1v1, GameModes1v1.TEST1v1];
+export const modesRatings = [GameModeRatings.TRIVIA, GameModeRatings.TRIVIA_TEMP]; // Trivia temp is used for ongoing games
 
 
 export const getGameModes = () => {
   return modesGame;
 };
 
 // Dont return test game mode
-export const getActiveGameModes = () => {
+export const getActiveGameModes = (): GameModesType[] => {
   const array = modesGame.slice();
-  array.splice(modesGame.indexOf('test'), 1);
-  array.splice(modesGame.indexOf('valorant'), 1);
-  return ['dota']; // Dota is default
+  array.splice(modesGame.indexOf(GameModesStandard.TEST), 1);
+  array.splice(modesGame.indexOf(GameModesStandard.VALORANT), 1);
```

Comment:
Should these be removed since they provide no functionality?
Also what about `GameModesStandard.CS`?

Reply:
Used the array and instead filtered out CS when temperature check fetches gamemodes